test/inst_update_partition_auto_test.rb

```diff
@@ -0,0 +1,207 @@
+#!/usr/bin/env rspec
+
+require_relative "test_helper"
+require "update/clients/inst_update_partition_auto"
+
+Yast.import "RootPart"
+Yast.import "Update"
+Yast.import "Installation"
+
+describe Yast::InstUpdatePartitionAutoClient do
+
+  describe "#main" do
+    let(:restarting) { false }
+
+    before do
+      stub_root_part
+      stub_const("Yast::FileSystems", double)
+      allow(Yast::Update)
+      allow(Yast::Installation).to receive(:restarting?) { restarting }
+      allow(Yast::Installation).to receive(:destdir).and_return("/mnt")
+      allow(Yast::Report).to receive(:error)
+      allow(Yast::Pkg).to receive(:TargetInitializeOptions)
+      allow(Yast::Pkg).to receive(:TargetFinish)
+      allow(Yast::Pkg).to receive(:TargetLoad).and_return(true)
+      stub_subject(subject)
+    end
+
+    context "when installation is restarting" do
+      let(:restarting) { true }
+
+      it "loads data from data dump file if present" do
+        expect(subject).to receive(:data_stored?).and_return(true)
+        expect(subject).to receive(:load_data)
+
+        expect(subject.main).to eql(:cancel)
```

Comment:
If you're checking really this value, then it's ok. But if you're just ignoring it, I would remove the expectation. But that's a matter of taste.